### DISCUSSION TITLE
element  Halo boxContent question

### DISCUSSION THREAD

**Discussion by james-ep01** [clientIO/joint#2834] [Q&A]
### Introduction

My element's objects show the appropriate property options in boxContent.
There are elements that do not have the property, so they return blank or null, and the screen displays a blank space with the theme color.

1. 1. Display appropriate properties for each element
![image](https://github.com/user-attachments/assets/e836c2f3-8a58-4894-8e7d-0e6e8d8051c1)

2. Show boxContent even when there is no property and boxContent is null or blank
![image](https://github.com/user-attachments/assets/d0b5df18-433a-4b75-b74b-d7bedab063d9)

My halo boxContent is as follows.

boxContent: function boxContent(cellView, boxDOMElement){
     var content = App.config.utils.getCellBoxHtml(cellView);
     return content;
},

How can I make it **not show** on the screen when boxContent is null or blank?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Currently, setting `false` (a boolean) is the only way to hide the Halo info box. If a callback is used for the `boxContent` option, the info box remains visible at all times.

**Reply by james-ep01** [reply]
Thank you.

It shouldn't be visible, but it keeps showing up on the screen and I wanted to get rid of it by selecting it. It kept showing up, so I ended up getting rid of it in a box style.

**Reply by kumilingus** [reply]
I am not sure I follow, but this what you should do:

```ts
const isVisible = condition === true;
new ui.Halo({
    /* ... */
    boxContent: isVisible ? (() => 'hello!') : false;
})
```

**Comment by james-ep01** [comment]
Thank you, it works well.

And v4.1 halo has Group feature and customize handle feature which is much more useful than the old pie way.

**Comment by james-ep01** [comment]
It does become invisible, but another problem arises.
When I change the inspector, the contents of the object's box do not change. Haha...
I think that if I use variables or values ​​outside the halo object, they do not render in the box.
I guess I should just get rid of them with style.

**Reply by kumilingus** [reply]
Could you provide an example?
